sklearn/preprocessing/tests/test_data.py

```diff
@@ -1201,53 +1210,59 @@ def test_quantile_transform_check_error():
                           [0, 0, 2.6, 4.1, 0, 0, 2.3, 0, 9.5, 0.1]])
     X_neg = sparse.csc_matrix(X_neg)
 
-    assert_raises_regex(ValueError, "Invalid value for 'n_quantiles': 0.",
-                        QuantileTransformer(n_quantiles=0).fit, X)
-    assert_raises_regex(ValueError, "Invalid value for 'subsample': 0.",
-                        QuantileTransformer(subsample=0).fit, X)
-    assert_raises_regex(ValueError, "The number of quantiles cannot be"
-                        " greater than the number of samples used. Got"
-                        " 1000 quantiles and 10 samples.",
-                        QuantileTransformer(subsample=10).fit, X)
+    with pytest.raises(ValueError, match="Invalid value for "
```

Comment:
a nicer way to be able to read it is to move the error message above `pytest.raises`

```python
err_msg = "Invalid value for 'n_quantiles':0."
with pytest.raises(ValueError, match=err_msg
)
```

You can even make a for loop (you should parametrize but let's keep for another PR)

```python
all_err_msg = ("Invalid value for 'n_quantiles': 0.",
               "Invalid value for 'subsample': 0.",
               ...)
all_params = ({'n_quantiles': 0.}, {'subsample': 0.}, {'subsample': 10})
for params, err_msg in zip(all_params, all_err_msg):
    with pytest.raises(ValueError, match=err_msg):
        QuantileTransformer(**params).fit(X)
```

Reply:
Thanks for the suggestion @glemaitre. Will move the errror msg out of `pytest.raises`. Will open another PR for parameterization purpose.